### PR TITLE
feat: adds trace id to error messages

### PR DESCRIPTION
**Pull request by mdarrik**
## Why are these changes introduced?

Addresses [ENG-8370](https://nacelle.atlassian.net/browse/ENG-8370) 

## What is this pull request doing?

- Adds the `x-amzn-trace-id` to the `response.error.message` field in the query response. This will allow users to provide info for datadog traces if they need to reach out for support.

## How to Test

1. Checkout this branch `mdarrik/ENG-8370-add-x-amzn-trace-id-to-errors`
2. Navigate to `packages/storefront-sdk`
3. Run `npm run coverage` - typechecking should pass and test coverage should be 100%
4. Run `npm run build` -  should build successfully
5. Use `npm link` to link to this build of the sdk in a test project. Run an invalid query. The `error.message` field should include `Trace Id: <trace-id>`.
6. (Optional) Use `npm link @nacelle/storefront-sdk` in the `packages/storefront-sdk-plugins/commerce-queries` folder. Then run `npm run test` all of the tests in that repo should continue to pass.

## Checklist

- [x] This Pull Request aligns with `nacelle-js`' [Code of Conduct](https://github.com/getnacelle/nacelle-js/blob/main/CODE_OF_CONDUCT.md#code-of-conduct).
- [x] You can follow your own "How to Test" instructions and get the expected result.
- [x] Screenshots, videos, etc. in the PR description are free of brand names and sensitive details.
- [x] Created a [changeset](https://github.com/changesets/changesets) (if the change should appear in a changelog).
- [x] Updated the `README.md` with documentation changes (if applicable).

[ENG-8370]: https://nacelle.atlassian.net/browse/ENG-8370?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/storefront-sdk/__mocks__/utils/getFetchPayload.ts [hunk 1]
```diff
@@ -1,11 +1,16 @@
 function getFetchPayload(
 	data: object,
-	responseOptions: { status: number } = {
+	responseOptions: { status: number; headers?: { [key: string]: string } } = {
```
Comment (NWRichmond) [inline packages/storefront-sdk/__mocks__/utils/getFetchPayload.ts:3]: Doesn't change the meaning at all, but to offer up an alternative typing:

```suggestion
	responseOptions: { status: number; headers?: Record<string, string> = {
```

File: .changeset/stale-cherries-compare.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@nacelle/storefront-sdk': minor
+---
+
+Adds a trace id to the `error` key in the response. This trace id should be included in any support requests
```
Comment (NWRichmond) [inline .changeset/stale-cherries-compare.md:5]:
```suggestion
Adds a trace id to the `error` key in the response. This trace id should be included in any support requests.
```

File: packages/storefront-sdk/src/client/index.test.ts [hunk 3]
```diff
@@ -22,6 +22,7 @@ import type {
 	SpaceProperties
 } from '../types/storefront.js';
 import type { StorefrontConfig } from '../types/config.js';
+import { GraphQLError } from 'graphql';
```
Comment (NWRichmond) [inline packages/storefront-sdk/src/client/index.test.ts:25]: nit: can we move this above the `import type`s?

File: packages/storefront-sdk/src/client/index.ts [hunk 4]
```diff
@@ -282,6 +282,17 @@ export class StorefrontClient {
 		return this.#graphqlClient
 			.query(query, variables as QVariables)
 			.toPromise()
+			.then(({ data, error }) => {
+				if (error) {
+					const traceHeader = (error.response as Response)?.headers?.get(
+						'x-amzn-trace-id'
+					);
+					if (traceHeader) {
+						error.message = `${error.message}\nTrace ID: ${traceHeader}. Please include this Trace ID in support requests`;
```
Comment (NWRichmond) [inline packages/storefront-sdk/src/client/index.ts:291]:
```suggestion
						error.message = `${error.message}\nTrace ID: ${traceHeader}. Please include this Trace ID in support requests.`;
```

**Review by NWRichmond — APPROVED** [review]
The tests pass, the package builds, and the `error.message` returned by the `query` method includes the Trace ID:

Storefront SDK error with Trace ID in error message

Nice!

I made some small suggestions, but nothing that merits a re-review.

**Review by KrisQ — APPROVED** [review]
Looks good!

**Pull request merged**